File: sdks/csharp/rest/src/Brume.Rest/Types/ApiKeyListResponse.cs
using Brume.Rest.Core;
using global::System.Text.Json;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

/// <summary>
/// `GET /v1/api-keys` success response.
/// </summary>
[Serializable]
public record ApiKeyListResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("api_keys")]
    public IEnumerable<ApiKeyListItem> ApiKeys { get; set; } = new List<ApiKeyListItem>();

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/rest/src/Types/ApiKeyListResponse.php
<?php

namespace Brume\Rest\Types;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;
use Brume\Rest\Core\Types\ArrayType;

/**
 * `GET /v1/api-keys` success response.
 */
class ApiKeyListResponse extends JsonSerializableType
{
    /**
     * @var array<ApiKeyListItem> $apiKeys
     */
    #[JsonProperty('api_keys'), ArrayType([ApiKeyListItem::class])]
    public array $apiKeys;

    /**
     * @param array{
     *   apiKeys: array<ApiKeyListItem>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->apiKeys = $values['apiKeys'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
